src/language_features/document_symbol.rs

```diff
@@ -23,6 +23,128 @@ use std::path::{Path, PathBuf};
 use unicode_width::UnicodeWidthStr;
 use url::Url;
 
+pub fn indent_guides(meta: EditorMeta, editor_params: EditorParams, ctx: &mut Context) {
+    let eligible_servers: Vec<_> = ctx
+        .language_servers
+        .iter()
+        .filter(|srv| attempt_server_capability(*srv, &meta, CAPABILITY_DOCUMENT_SYMBOL))
+        .collect();
+    let req_params = eligible_servers
+        .into_iter()
+        .map(|(server_name, _)| {
+            (
+                server_name.clone(),
+                vec![DocumentSymbolParams {
+                    text_document: TextDocumentIdentifier {
+                        uri: Url::from_file_path(&meta.buffile).unwrap(),
+                    },
+                    partial_result_params: Default::default(),
+                    work_done_progress_params: Default::default(),
+                }],
+            )
+        })
+        .collect();
+    let params = IndentGuidesParams::deserialize(editor_params).unwrap();
+    ctx.call::<DocumentSymbolRequest, _>(
+        meta,
+        RequestParams::Each(req_params),
+        move |ctx: &mut Context, meta, results| {
+            match results.into_iter().find(|(_, v)| v.is_some()) {
+                Some((server_name, Some(DocumentSymbolResponse::Nested(symbols))))
+                    if !symbols.is_empty() =>
+                {
+                    let server = &ctx.language_servers[&server_name];
+                    let mut range_specs = String::default();
+                    indent_guides_calc_range_specs(
+                        &symbols,
+                        &params,
+                        &meta,
+                        ctx,
+                        server,
+                        &mut range_specs,
+                        0,
+                    );
+                    let version = meta.version;
+                    let command = formatdoc!(
+                        "set-option buffer lsp_indent_guides {version} {range_specs}
+                         set-option buffer lsp_indent_guides_timestamp {version}"
+                    );
+                    let command = format!(
+                        "evaluate-commands -buffer {} -- {}",
+                        editor_quote(&meta.buffile),
+                        editor_quote(&command)
+                    );
+                    ctx.exec(meta, command);
+                }
+                _ => {
```

Comment:
maybe better to explicitly ignore `DocumentSymbolResponse::Flat` here, so the compiler will tell us if there's a new enum variant

Reply:
Yes, done!